lib/matplotlib/patches.py

```diff
@@ -416,14 +422,37 @@ def set_color(self, c):
         Patch.set_facecolor, Patch.set_edgecolor
             For setting the edge or face color individually.
         """
-        self.set_facecolor(c)
         self.set_edgecolor(c)
+        self.set_hatchcolor(c)
+        self.set_facecolor(c)
+
+    def _set_hatchcolor(self, color):
+        color = mpl._val_or_rc(color, 'hatch.color')
+        if color == 'edge':
+            self._hatch_color = 'edge'
+        else:
+            self._hatch_color = colors.to_rgba(color, self._alpha)
+        self.stale = True
+
+    def set_hatchcolor(self, color):
+        """
+        Set the patch hatch color.
+
+        Parameters
+        ----------
+        color : :mpltype:`color` or 'edge' or None
+        """
+        if isinstance(color, str) and color.lower() == 'edge':
```

Comment:
```suggestion
        if cbook._str_lower_equal(color, 'edge'):
```

Reply:
```suggestion
        if cbook._str_equal(color, 'edge'):
```
Let's not be permissive here. enforcing a single way of writing increases consistency in user code. Also the typing is limited to `Literal["edge"]`.